### ISSUE TITLE
Locale(fr) + moment('maintenance', 'DD MMMM YYYY') gives a valid date

### ISSUE THREAD

**Issue by msieurtoph** [moment/moment#3982]
**Description of the Issue and Steps to Reproduce:**

```js
require('moment').locale('fr');
moment('foobar', 'DD MMMM YYYY').isValid(); // gives false, which is expected
moment('maintenance', 'DD MMMM YYYY').isValid(); // gives true, which is NOT expected.
```
For the records, I also noticed that 

```js
moment('maintenant', 'DD MMMM YYYY').isValid() //gives true too. ('Maintenant' means 'now' in french, and it is close to 'maintenance')

// and 
moment('maintenance', 'DD MMM YYYY').isValid() // true
// but
moment('maintenance', 'DD MM YYYY').isValid() // false
```
And when I do not use the 'fr' locale, I get expected falsy results.
I guess there are some mistakes when using 'fr' locale, a string close to 'maintenant' and a certain 'month' pattern for decoding...  
Any idea ?

Thx in advance.
**Environment:**

```
$ node -v
v7.0.0
$ npm -v
4.5.0
```
**Other information**

```js
console.log( (new Date()).toString()) // Mon May 29 2017 09:35:15 GMT+0200 (Paris, Madrid (heure d’été))
console.log((new Date()).toLocaleString()) // 2017-05-29 09:35:15
console.log( (new Date()).getTimezoneOffset()) // -120
// console.log( navigator.userAgent)
console.log(moment.version) // 2.18.1
```

**Comment by icambron** [comment]
Moment has no idea what strings like "now" and "maintenant" mean. In any case, such a string is certainly not 'DD MMM YYYY'. What's happening is that Moment's parser is by default very aggressive about trying to extract some information from the string. We usually tell people to add the `strict` argument, like this:

```js
moment('maintenant', 'DD MMMM YYYY', true).isValid() //=> false
```

Without it, here what's happening:

 1. I can't find a digit to match DD, so skip that
 2. Ooh, "mai" is a French month name that matches "MMM" (MMM is for abbreviated names but "mai" is also the abbreviated form of mai)
 3. I don't know what to do with 'ntenant' so skip that.
 4. The net result is today's date and time but in May. Since it *is* May, that's actually just now

The same logic applies equally to "maintenance", it does the same thing there too.

TLDR: Moment doesn't support strings like "maintenant". Use strict parsing so it doesn't give you confusing behavior.

**Comment by msieurtoph** [comment]
Yeah.
I missed the "strict" part.
Thanks for the complete explanation about the "what's happening"...
